packages/contracts-bedrock/src/L2/L2ToL2CrossDomainMessenger.sol

```diff
@@ -99,15 +100,26 @@ contract L2ToL2CrossDomainMessenger is IL2ToL2CrossDomainMessenger, ISemver, Tra
     /// @param _destination Chain ID of the destination chain.
     /// @param _target      Target contract or wallet address.
     /// @param _message     Message payload to call target with.
-    function sendMessage(uint256 _destination, address _target, bytes calldata _message) external {
+    /// @return _msgHash The hash of the message being sent, which can be used for tracking whether
+    ///                  the message has successfully been relayed.
+    function sendMessage(
+        uint256 _destination,
+        address _target,
+        bytes calldata _message
+    )
+        external
+        returns (bytes32 _msgHash)
+    {
         if (_destination == block.chainid) revert MessageDestinationSameChain();
         if (_target == Predeploys.CROSS_L2_INBOX) revert MessageTargetCrossL2Inbox();
         if (_target == Predeploys.L2_TO_L2_CROSS_DOMAIN_MESSENGER) revert MessageTargetL2ToL2CrossDomainMessenger();
 
+        (uint256 source, uint256 nonce, address sender) = (block.chainid, messageNonce(), msg.sender);
```

Comment:
```suggestion
        (uint256 _source, uint256 _nonce, address _sender) = (block.chainid, messageNonce(), msg.sender);
```
we tend to use underscored variables as to avoid confusing them with storage ones.

Reply:
hmm I think that conflicts with @tynes comment here: https://github.com/ethereum-optimism/optimism/pull/12089#discussion_r1774017648

Reply:
oh lol... then discard my comment :+1: